src/lib/Form/TrashLocationOptionProvider/HasChildren.php

```diff
@@ -34,10 +41,15 @@ public function supports(Location $location): bool
 
     public function addOptions(FormInterface $form, Location $location): void
     {
-        $childCount = $this->locationService->getLocationChildCount($location);
+        $limit = $this->configResolver->getParameter('subtree_operations.query_subtree.limit');
+
+        $useLimit = $limit > 0;
```

Comment:
Ideally we would be more specific, and use `null` value to indicate limitless queries - and treat non-positive/null values in parameter as an error.